server/core/src/main/scala/org.apache.daffodil.tdml/TDML.scala

```diff
@@ -0,0 +1,214 @@
+  // tdmlDescription: Description for the DFDL operation
+  // tdmlPath:        Path to the TDML file
+  def append(
+      infosetPath: String,
+      schemaPath: String,
+      dataPath: String,
+      tdmlName: String,
+      tdmlDescription: String,
+      tdmlPath: String
+  ): Unit = {
+
+    val testSuite = JAXBContext
+      .newInstance(classOf[TestSuite])
+      .createUnmarshaller()
+      .unmarshal(new File(tdmlPath))
+      .asInstanceOf[TestSuite]
+
+    testSuite
+      .getTutorialOrParserTestCaseOrDefineSchema()
+      .add(createTestCase(infosetPath, schemaPath, dataPath, tdmlName, tdmlDescription))
+
+    val marshaller = JAXBContext.newInstance(classOf[TestSuite]).createMarshaller()
+    marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true)
+    marshaller.marshal(testSuite, new java.io.File(tdmlPath))
+  }
+
+  // Find the parameters needed to execute a DFDL parse based on the given TDML Parameters
+  //
+  // tdmlName:        Test case name to run
+  // tdmlDescription: Description of test case to run
+  // tdmlPath:        File path of TDML file to extract test case from
+  //
+  // Returns a tuple containing the following (Path to DFDL Schema, Path to Data File)
+  // All paths returned could be either relative or absolute - it depends on what exists in the TDML file
+  def execute(tdmlName: String, tdmlDescription: String, tdmlPath: String): (String, String) = {
+    val basePath = Paths.get(tdmlPath).toAbsolutePath().getParent()
+
+    val testCaseList = JAXBContext
+      .newInstance(classOf[TestSuite])
+      .createUnmarshaller()
+      .unmarshal(new File(tdmlPath))
+      .asInstanceOf[TestSuite]
+      .getTutorialOrParserTestCaseOrDefineSchema()
+
+    testCaseList.forEach { tc =>
+      tc match {
+        case ptc: ParserTestCaseType =>
+          if (ptc.getName() == tdmlName && ptc.getDescription() == tdmlDescription) {
+            ptc.getTutorialOrDocumentOrInfoset().forEach { dis =>
+              dis match {
+                case doc: DocumentType =>
+                  // The right part of the tuple only takes the first DocumentPart inside the Document.
+                  // In the case that there are more than one, any extras will be ignored.
+                  val schemaPath = Paths.get(basePath + File.separator + ptc.getModel()).normalize().toString()
+                  val dataPath = Paths
+                    .get(
+                      basePath + File.separator + doc
+                        .getContent()
+                        .get(0)
+                        .asInstanceOf[JAXBElement[DocumentPartType]]
+                        .getValue()
+                        .getValue()
+                    )
+                    .normalize()
+                    .toString()
+                  return (schemaPath, dataPath)
+              }
+            }
+          }
+      }
+    }
+
+    // If there is no test case in the TDML file meeting the name/description criteria, return empty
```

Comment:
Suggestion: the comment implies that the return type should be `Option[(String, String)]`.

Reply:
Does rephrasing the end portion from "... return empty" to "... return an empty tuple" remove this implication?

Reply:
Sorry, I mean I'm actually suggesting changing the return type to `Option[(String, String)]` so the caller doesn't have to check for empty strings.